command/agent/rpc.go

```diff
@@ -616,6 +626,25 @@ func (i *AgentRPC) handleSerfQuery(client *rpcClient, seq uint64) error {
 	return client.Send(&resp, nil)
 }
 
+func (i *AgentRPC) handleSerfPing(client *rpcClient, seq uint64) error {
+	var req serfPingRequest
+	if err := client.dec.Decode(&req); err != nil {
+		return fmt.Errorf("decode failed: %v", err)
+	}
+
+	i.logger.Printf("[DEBUG] agent.rpc: handling serf ping request")
+
+	// Start the serf query
+	resp, err := i.agent.SerfPing(req.Name)
```

Comment:
Looking closer at the code, the existing pattern can also be found in handleForceLeave() and handleJoin().  The suggestion I had for line 645 cannot be found elsewhere.  I think with this observation I would prefer to not modify the code so as to follow the precedent set by other implementations.